### PR TITLE
Fix converting chat to project

### PR DESCRIPTION
**Pull request by durulkoca**

### CODE REVIEW

File: src/widgets/Chat/useChat.jsx [hunk 1]
```diff
@@ -71,23 +71,24 @@ const useChat = () => {
...
+        name: "Chat Project",
+        description: "This project has been converted from chat",
       },
     };
 
-    localStorage.setItem(`ide.projects.${id}`, JSON.stringify(project));
+    localStorage.setItem(`ide.context.${id}`, JSON.stringify(project));
     publish("CHAT_CONVERTED", chat);
-    navigate(`/${id}/api?mode=local`);
     navigate(0);
```
Comment (canmingir) [inline src/widgets/Chat/useChat.jsx:90]: Can we do it without `navigate(0)`? We had an issue before, but it should be working now

File: src/lib/test/Project.test.js [hunk 2]
```diff
@@ -1,73 +1,50 @@
 import Project from "../Project.js";
+import chatData from "./chatData.json";
```
Comment (canmingir) [inline src/lib/test/Project.test.js:2]: Can you clean up examples that won't have any chat ref?

File: src/lib/test/Project.test.js [hunk 3]
```diff
@@ -1,73 +1,50 @@
...
+  );
+
+  const result = Project.compile([imperativeBlock]);
 
-  expect(result.api.length).toBe(0);
-  expect(result.functions.length).toBe(0);
-  expect(result.declarations.length).toBe(0);
+  expect(result.functions).toEqual([]);
+  expect(result.declarations).toEqual([]);
 });
```
Comment (canmingir) [inline src/lib/test/Project.test.js:50]: Can you add additional test case that having imperative block of code without `"use imperative"`?

**Review by canmingir — CHANGES_REQUESTED** [review]

File: src/widgets/Chat/useChat.jsx (hunk 1 above)
Comment (durulkoca) [reply src/widgets/Chat/useChat.jsx]: solved using startTransition

File: src/lib/test/Project.test.js (hunk 2 above)
Comment (durulkoca) [reply src/lib/test/Project.test.js:2]: i cleaned up the user messages. can you check again?

File: src/lib/test/Project.test.js (hunk 3 above)
Comment (durulkoca) [reply src/lib/test/Project.test.js:50]: added an additional test case

**Review by canmingir — APPROVED** [review]

**Pull request merged**